src/Alpha_complex/include/gudhi/Alpha_complex.h

```diff
@@ -535,6 +535,21 @@ class Alpha_complex {
       }
     }
   }
+
+ public:
+  // Kernel::Point_d are not weighted points
+  std::vector<typename Kernel::Point_d> get_point_cloud() {
+    std::vector<typename Kernel::Point_d> point_cloud;
+    int point_cloud_size = vertex_handle_to_iterator_.size();
+    point_cloud.resize(point_cloud_size);
+    for (int index = 0; index < point_cloud_size; index++) {
```

Comment:
`std::transform` + `std::back_inserter` on 0c21c365f

Reply:
That's equivalent to option 2, except that you forgot `reserve()`.

Reply:
I added `reserve()` on 9d1985c6e but it seems that I misunderstood the 3rd point. To be discussed.

Reply:
3rd version was using `point_cloud.insert` (or even the equivalent constructor), which takes begin and end iterators. Since we don't want to copy the elements of vertex_handle_to_iterator_, we can't use directly its begin/end pointers, we have to go through boost::adaptors::transform, boost::transform_iterator, or something similar, after checking that the size of the transformed range still takes constant time to read.

The drawback of option 1 is that resize writes 0 (or some default value) in the whole vector before the loop starts. Option 2 has some overhead because each insertion checks the capacity and changes the size. Option 3 supposedly avoids both. (a clever compiler might be able to remove the overhead of 1 or 2, but last time I checked it wasn't happening)

But again, I was ok with your first version (or the new one).